api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,578 @@
+        """
+        Private method for network checks.
+
+        :param args: network specific checking parameters and arguments
+        :return:
+        """
+        res: List[CheckEntry] = list()
+
+        minion_id = local_minion_id()
+
+        cluster_path = KeyPath('cluster')
+        node_network_path = KeyPath(f'cluster/{minion_id}/network/data_nw')
+
+        pillar_keys = (PillarKey(cluster_path / 'type'),
+                       PillarKey(node_network_path / "public_ip_addr"),
+                       PillarKey(node_network_path / "pvt_ip_addr"),
+                       PillarKey(cluster_path / "cluster_ip"))
+
+        # maybe we need to call it from the all targets and analyze the output
+        # from each node
+        pillar = PillarResolver(cfg.LOCAL_MINION).get(pillar_keys)
+
+        pillar = pillar.get(local_minion_id())  # type: dict
+
+        for key in pillar_keys:
+            _res = CheckEntry(cfg.Checks.NETWORK.value)
+            if not pillar[key] or pillar[key] is values.MISSED:
+                _res.set_fail(target=str(key.keypath),
+                              comment="value is not specified")
+            elif pillar[key] is None:
+                _res.set_fail(target=str(key.keypath),
+                              comment="value is unset")
+            else:
+                _res.set_passed(target=str(key.keypath))
+
+            res.append(_res)
+
+        return res
+
+    @staticmethod
+    def _communicability(
+            *, args: str,
+            targets: Union[list, tuple, set] = PENDING_SERVERS) -> CheckEntry:
+        """
+        Check if nodes can communicate using `salt test.ping`
+
+        :param args: bmc accessibility check specific parameters and arguments
+        :param targets: target nodes where network checks will be executed
+        :return:
+        """
+        res: CheckEntry = CheckEntry(cfg.Checks.COMMUNICABILITY.value)
+
+        # TODO: need to resolve cfg.ALL_MINIONS alias
+        #  to the list of minions
+        # TODO: create check and return list of nodes which are down
+        if check_salt_minions_are_ready(targets=targets):
+            res.set_passed(target=local_minion_id())
+        else:
+            res.set_fail(target=local_minion_id())
+
+        return res
+
+    @staticmethod
+    def _bmc_accessibility(*, args: str) -> CheckEntry:
+        """
+        Check BMC accessibility
+
+        :param args: bmc accessibility check specific parameters and arguments
+        :return:
+        """
+        _BMC_CHECK_CMD = 'ipmitool chassis status | grep "System Power"'
+
+        res: CheckEntry = CheckEntry(cfg.Checks.BMC_ACCESSIBILITY.value)
+
+        minion_id = local_minion_id()
+
+        try:
+            _res = cmd_run(_BMC_CHECK_CMD, targets=minion_id,
+                           fun_kwargs=dict(python_shell=True))
+            # TODO: parse command output if necessary
+            res.set_passed(target=minion_id, comment=str(_res))
```

Comment:
better to use `else` block for that since an exception inside `set_passed` will lead to `set_fail` call which will shift an attention internal logic to a check fail

Reply:
Done